install.js

```diff
@@ -0,0 +1,52 @@
+const { execSync, spawn, spawnSync } = require('child_process');
+const { existsSync } = require('fs');
+const readline = require('readline');
+
+const isVersionGreaterThan = (version, targetMajor) => Number(version.split('.')[0]) >= targetMajor;
+
+const trimV = (version) => version.startsWith('v') ? version.slice(1) : version;
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const directory = 'logto';
+
+(async () => {
+  if (existsSync(directory)) {
+    throw new Error(`\`${directory}\` already exists in current directory.`);
+  }
+
+  const nodeVersion = execSync('node -v', { encoding: 'utf-8' });
+  const pgOutput = execSync('postgres --version', { encoding: 'utf-8' });
+
+  if (!isVersionGreaterThan(trimV(nodeVersion), 16)) {
+    throw new Error('Logto requires NodeJS>=16.0.0');
+  }
+
+  const pgArray = pgOutput.split(' ');
+  const pgVersion = pgArray[pgArray.length - 1];
+
+  if (!isVersionGreaterThan(trimV(pgVersion), 14)) {
+    throw new Error('Logto requires PostgreSQL>=14.0.0');
```

Comment:
Do we need a more specific name here, `PostgreSQL Client` is what we need, if can be confusing for some users.

Reply:
Would it be possible that we also print out the command that installs the required package?

Reply:
@wangsijie shoot, you are right. Postgres is not necessary actually. let me update this
@demonzoo it's hard since we got lots of ways to install according to specific OS